Valikatselmuksen kiirelliset

jpluomajoki-solita opened the pull request.

jpluomajoki-solita committed:
Formatoi yhteenvedon numerot euroiksi

jpluomajoki-solita committed:
korjaa loput numerot euromuotoisiksi

jpluomajoki-solita committed:
Päivitä oikaisujen luokat

jpluomajoki-solita committed:
Indeksikorjatut arvot välikatselmukseen

topipoh committed:
Estä välikatselmuksen radio buttonien hajoaminen kapealla näytöllä

jpluomajoki-solita committed:
Lisää välikatselmusnappi sivun yläreunaan

jpluomajoki-solita committed:
Siisti koodia

jpluomajoki-solita committed:
Korjaa päätösten validointi

jpluomajoki-solita committed:
Migraatio kattohinnan oikaisuille

topipoh committed:
Lisää taulumääritys valikatselmus/kattohinnan-oikaisu

topipoh committed:
Lisää hae/tallenna/poista -endpointit kattohinnan oikaisuille (WIP)

jpluomajoki-solita committed:
nimeä hoitokausi->hoitokauden-alkuvuosi

topipoh committed:
Määrittele kattohinnan_oikaisu.poistettu oletusarvoksi false

topipoh committed:
Toteuta hae/tallenna -endpointit kattohinnan oikaisuille

topipoh committed:
Merge remote-tracking branch 'origin/valikatselmuksen-kiirelliset' into valikatselmuksen-kiirelliset

# Conflicts:
#	tietokanta/src/main/resources/db/migration/V1_999__kattohinnan-oikaisu.sql

jpluomajoki-solita committed:
Oikaistu kattohinta kustannussuunnitelmaan

topipoh committed:
Lisää CONSTRAINT uniikki_urakka_alkuvuosi

topipoh committed:
Lisää input-kenttä ja nappi kattohinnan oikaisua varten

topipoh committed:
Merge remote-tracking branch 'origin/valikatselmuksen-kiirelliset' into valikatselmuksen-kiirelliset

jpluomajoki-solita committed:
Oikaistu kattohinta valikatselmukseen

topipoh committed:
Lisää label kattohinta-kentälle

topipoh committed:
Tee kentästä leveämpi

topipoh committed:
Merge remote-tracking branch 'origin/valikatselmuksen-kiirelliset' into valikatselmuksen-kiirelliset

topipoh committed:
Määrittele max desimaalit ja kokonaisosan määrä niin että mahtuu doubleen ilman virhettä

jpluomajoki-solita committed:
Korjaa oikaistu kattohinta kustannussuunnitelmassa

jpluomajoki-solita committed:
Validoi päätökset oikaistulla kattohinnalla

topipoh committed:
Mahdollista kattohinnan oikaisun muokkaaminen tallennuksen jälkeen

topipoh committed:
Merge remote-tracking branch 'origin/valikatselmuksen-kiirelliset' into valikatselmuksen-kiirelliset

topipoh committed:
Poista päätökset kattohinnan oikaisun jälkeen

topipoh committed:
Vie eventiä TallennaKattohinnanOikaisuOnnistui oikeampaan suuntaan

jpluomajoki-solita committed:
Oikaistu kattohinta kust.suun. yhteenvetoon.

 - Korjaa myös tavoitehinnan oikaisulla oikaistu tavoite- ja kattohinta

topipoh committed:
Mahdollista kattohinnan oikaisun poistaminen ja poiston kumoaminen

jpluomajoki-solita committed:
Korjaile oikaistuja numeroita, hae uusi budjettisuunnitelma useammin

topipoh committed:
Mahdollista käyttöliittymästä kattohinnan oikaisun poistaminen

topipoh committed:
Nollaa tietokanta jokaisen valikatselmus-testin välissä, koska vaikuttavat toisiinsa

topipoh committed:
Lisää oikeuksiin liittyvä testi

topipoh committed:
Tyhjennä :kattohinnan-oikaisu -lomake

topipoh committed:
Palauta kattohinnan oikaisut muodossa vuosi->oikaisu (oli vuosi->oikaisut)

topipoh committed:
Lisää docstring

topipoh committed:
Tyhjennä :kattohinnan-oikaisu -lomake kun haetaan oikaisut

topipoh committed:
Määritä pvm/nyt jotta tulos ei riipu testien ajamishetkestä

topipoh committed:
Tarkista urakkatyyppi teiden-hoito

topipoh committed:
Tee kattohinnan oikaisu transaktion sisällä

topipoh committed:
Tee kattohinnan oikaisun poisto transaktion sisällä

jpluomajoki-solita committed:
Merge branch 'valikatselmuksen-kiirelliset' of github.com:finnishtransportagency/harja into valikatselmuksen-kiirelliset

jpluomajoki-solita committed:
Korjaile välikatselmuksen tekstejä

jpluomajoki-solita committed:
Korjaa välikatselmuksen numeroita

jpluomajoki-solita committed:
Koodin siistimistä

jpluomajoki-solita committed:
Nimeä hankintakustannukset oikein

topipoh committed:
Tarkista, että oikaistu kattohinta on suurempi kuin oikaistu tavoitehinta

topipoh committed:
Merge remote-tracking branch 'origin/valikatselmuksen-kiirelliset' into valikatselmuksen-kiirelliset

topipoh committed:
Käytä riittävän isoa kattohintaa testeissä

topipoh committed:
Lisää tarkistuksia

topipoh committed:
Tarkista käyttöliittymällä, että kattohinta on suurempi kuin tavoitehinta. Lisää muokkaa ja poista -napit.

topipoh committed:
Merge remote-tracking branch 'origin/valikatselmuksen-kiirelliset' into valikatselmuksen-kiirelliset

jpluomajoki-solita committed:
Näytä indeksikorjatut arvot kust. seurannassa

jpluomajoki-solita committed:
Korjaa bugi, pos-int? -> pos?

jpluomajoki-solita committed:
Poista vanhentunut fixme-kommentti

 - Kyseinen sarake muutettiin hoitokausi->hoitokauden-alkuvuosi

jpluomajoki-solita committed:
Nimeä migraatio oikeaksi

jpluomajoki-solita committed:
Merge branch 'develop' into valikatselmuksen-kiirelliset

K1ll3rF0x commented on src/cljs/harja/views/urakka/suunnittelu/kustannussuunnitelma/kustannussuunnitelma_view.cljs:
```diff
@@ -38,8 +38,10 @@
+  [osio-avain summien-maara m]
+  [:div
+   {:class ["sisalto"
+            (when (and (= ::t/tavoite-ja-kattohinta osio-avain) (= 6 summien-maara)) "tavoite-ja-kattohinta")]}
```
Käy väliaikaiseksi ratkaisuksi. Refaktoroidaan myöhemmin.

K1ll3rF0x commented on src/cljs/harja/views/urakka/kulut/valikatselmus.cljs:
```diff
@@ -115,7 +162,13 @@
+                                               (seq @oikaisut-atom)
+                                               voi-muokata?
+                                               ;; FIXME: ota pois kommenteista
+                                               ;(lupaus-domain/vuosi-19-20? urakan-alkuvuosi)
```
Ota pois kommenteista

rckopteri committed:
Siirrä rahavaraukset hankintakustannusten alta rahavaraukset pääryhmän alle.

K1ll3rF0x commented on src/cljs/harja/views/urakka/suunnittelu/kustannussuunnitelma/yhteiset.cljs:
```diff
@@ -376,7 +376,9 @@
-                              (e! (tuck-apurit/->MuutaTila [:suodattimet :hoitokauden-numero] hoitovuosi)))
+                              (do
+                                (e! (tuck-apurit/->MuutaTila [:suodattimet :hoitokauden-numero] hoitovuosi))
+                                (e! (t/->HaeBudjettitavoite))))
```
Miksi budjettitavoite pitää hakea joka kerta uudestaan, kun hoitovuotta vaihtaa?

jpluomajoki-solita replied:
Jäänyt kehitysvaiheesta, ei tarpeellinen.

topipoh committed:
Näytä kattohinnan oikaisu vain 19-20 -urakoille

topipoh committed:
Levennä yhteenvetolaatikkoa, ettei layout mene rikki

jpluomajoki-solita committed:
Korjaa js-numerobugi, piilota virheellisesti näytetty tekstilaatikko

jpluomajoki-solita committed:
Piilota oikaistut hinnat jos ei oikaisuja kustannussuunnitelmasta

jpluomajoki-solita committed:
Nimeä suunnitellut hankintakustannukset->hankinnat

topipoh committed:
Älä joinaa väärien hoitovuosien tavoitehinnan oikaisuja hae-budjettitavoite -kyselyssä

topipoh committed:
Refaktoroi hae-budjettitavoite -kyselyä (alikysely erikseen)

jpluomajoki-solita committed:
Säilytä päätösten lomakkeet jos urakka ja hoitokausi pysyy samana.

 - Välikatselmuksen lomakkeet säilyttävät täytetyt tiedot, jos sivua
   vaihtaa, mutta hoitokausi ja valittu urakka pysyy samana.

topipoh committed:
Pyöristetään liukulukujen vähennyslaskun tulos

topipoh committed:
Merge remote-tracking branch 'origin/valikatselmuksen-kiirelliset' into valikatselmuksen-kiirelliset

jpluomajoki-solita commented on src/cljs/harja/views/urakka/kulut/valikatselmus.cljs:
```diff
@@ -146,7 +198,7 @@
             :nimi ::valikatselmus/otsikko
             :tyyppi :valinta
-            :valinnat valikatselmus/luokat
+            :valinnat valikatselmus/luokat ;; TODO: Älä näytä "Alleviivatun fontin vaikutus tavoitehintaan" muissa kuin 19-20 alkaneissa.
```
Korjattavat TODO, ei räjäytä harjaa jos muut näkevän tämän nyt

topipoh committed:
Lasketaan indeksikorjattu kattohinta myös 2019 ja 2020 urakoille

topipoh committed:
Korjaa viimeisin_migraatio.sh toimimaan mistä tahansa polusta ajettuna

topipoh committed:
Nosta migraation versionumero 904->907

topipoh committed:
Merge remote-tracking branch 'origin/develop' into valikatselmuksen-kiirelliset

jpluomajoki-solita committed:
Salli menneiden hoitokausien välikatselmus 19-20 urakoille

topipoh committed:
Merge branch 'develop' into valikatselmuksen-kiirelliset
